### PR TITLE
Feat/attrition table histogram modal

### PR DESCRIPTION
**Pull request by jarvisraymond-uchicago**
Link to JIRA ticket if there is one: 

https://ctds-planx.atlassian.net/browse/VADC-246
https://ctds-planx.atlassian.net/browse/VADC-721

**Implementation**
Continuous Outcome:
output
Dichotomous Outcome: 
output
Storybooks: 
output

### New Features
* Adds modal to attrition table in GWAS App for histogram and euler diagram visualizations. Modal can be triggered using chart icon.

### CODE REVIEW

File: src/Analysis/GWASApp/Components/AttritionTableWrapper/AttritionTable/AttritionTableModal/AttritionTableModal.jsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+            }}
+            selectedControlCohort={{
+              cohort_name: modalInfo?.rowObject?.cohort_names[1],
+              size: modalInfo?.rowObject?.cohort_sizes[1],
+              cohort_definition_id: modalInfo?.rowObject?.cohort_ids[1],
+            }}
+            // Need to remove first item from currentCovariateAndCovariatesFromPrecedingRows for this visualization
+            selectedCovariates={modalInfo.currentCovariateAndCovariatesFromPrecedingRows.slice(
+              1,
+            )}
```
Comment (pieterlukasse) [inline src/Analysis/GWASApp/Components/AttritionTableWrapper/AttritionTable/AttritionTableModal/AttritionTableModal.jsx:55]: as discussed, this removal should only happen for the visualization of the "outcome" row diagram

File: src/Analysis/GWASApp/Components/AttritionTableWrapper/AttritionTable/AttritionTableRow.jsx [hunk 2]
```diff
@@ -130,13 +132,31 @@ const AttritionTableRow = ({
     return value;
   };
 
+  const determineModalTitle = () => {
+    let title = rowObject.variable_type === 'concept' ? 'Continuous ' : 'Dichotomous ';
+    title += rowType;
+    title += rowType === 'Outcome' ? ' Phenotype' : '';
+    return title;
+  };
+  const handleChatIconClick = () => {
```
Comment (pieterlukasse) [inline src/Analysis/GWASApp/Components/AttritionTableWrapper/AttritionTable/AttritionTableRow.jsx:141]:
```suggestion
  const handleChartIconClick = () => {
```

File: src/Analysis/GWASApp/Components/Covariates/CustomDichotomousCovariates.jsx [hunk 3]
```diff
@@ -21,6 +21,14 @@ const CustomDichotomousCovariates = ({
...
     const dichotomous = {
       variable_type: 'custom_dichotomous',
+      cohort_names: [
+        firstPopulation.cohort_name,
+        secondPopulation.cohort_name,
+      ],
+      cohort_sizes: [
+        firstPopulation.size,
+        secondPopulation.size,
+      ],
```
Comment (pieterlukasse) [inline src/Analysis/GWASApp/Components/Covariates/CustomDichotomousCovariates.jsx:31]: why this change?

**Review by pieterlukasse — COMMENTED** [review]
thanks @jarvisraymond-uchicago . Please see the comments below.

File: src/Analysis/GWASApp/Components/Covariates/CustomDichotomousCovariates.jsx (hunk 3 above)
Comment (jarvisraymond-uchicago) [reply src/Analysis/GWASApp/Components/Covariates/CustomDichotomousCovariates.jsx:31]: The Euler diagram requires sizes and labels (key value pairs with key names .cohort_name and .size) in order to render the visualization. Before the attrition table didn't need to render a Euler diagram so this information wasn't stored, only the  cohort_ids were stored. To minimize refactoring, the additional information is now stored in arrays the same way the cohort_ids are stored.

File: src/Analysis/GWASApp/Components/AttritionTableWrapper/AttritionTable/AttritionTableModal/AttritionTableModal.jsx (hunk 1 above)
Comment (jarvisraymond-uchicago) [reply src/Analysis/GWASApp/Components/AttritionTableWrapper/AttritionTable/AttritionTableModal/AttritionTableModal.jsx]: This has been updated to use a variable to determine if the row is the outcome:

` const rowIsOutcome =  modalInfo.currentCovariateAndCovariatesFromPrecedingRows?.length === 1;`

Then the selectedCovariates and outcome props for the CohortsOverlapDiagram use this variable to determine the values: 

```
            selectedCovariates={rowIsOutcome ? [] : modalInfo.currentCovariateAndCovariatesFromPrecedingRows}
            outcome={rowIsOutcome ? null : modalInfo.outcome}
```

File: src/Analysis/GWASApp/Components/AttritionTableWrapper/AttritionTable/AttritionTableRow.jsx (hunk 2 above)
Comment (jarvisraymond-uchicago) [reply src/Analysis/GWASApp/Components/AttritionTableWrapper/AttritionTable/AttritionTableRow.jsx:141]: Good catch! Thank you - updated now

File: src/Analysis/GWASApp/Components/AttritionTableWrapper/AttritionTable/AttritionTable.jsx [hunk 4]
```diff
@@ -95,6 +114,8 @@ const AttritionTable = ({
                     rowType='Cohort'
                     rowObject={null}
                     currentCovariateAndCovariatesFromPrecedingRows={[]}
+                    modalInfo={modalInfo}
+                    setModalInfo={setModalInfo}
```
[the remark this answers is not in the document]
Comment (jarvisraymond-uchicago) [reply src/Analysis/GWASApp/Components/AttritionTableWrapper/AttritionTable/AttritionTable.jsx:118]: Good point, this has been removed and default values have been set for the AttritionTableRow component 

```
  modalInfo: null,
  setModalInfo: ()=> null,
```

File: src/Analysis/GWASApp/Components/AttritionTableWrapper/AttritionTableModal/AttritionTableModal.jsx [hunk 5]
```diff
@@ -49,17 +50,8 @@ const AttritionTableModal = ({ modalInfo, setModalInfo }) => {
...
-            )}
-            // Outcome should be null if we're viewing the outcome for this visualization
-            outcome={
-              modalInfo.currentCovariateAndCovariatesFromPrecedingRows
-                .length === 1
-                ? null
-                : modalInfo.outcome
-            }
+            selectedCovariates={rowIsOutcome ? [] : modalInfo.currentCovariateAndCovariatesFromPrecedingRows}
+            outcome={rowIsOutcome ? null : modalInfo.outcome}
```
Comment (pieterlukasse) [inline src/Analysis/GWASApp/Components/AttritionTableWrapper/AttritionTableModal/AttritionTableModal.jsx:54]: nice!

File: src/Analysis/GWASApp/Components/AttritionTableWrapper/AttritionTableModal/AttritionTableModal.jsx [hunk 6]
```diff
@@ -2,11 +2,12 @@ import React from 'react';
...
 import { Modal } from 'antd';
 import './AttritionTableModal.css';
-import PhenotypeHistogram from '../../../Diagrams/PhenotypeHistogram/PhenotypeHistogram';
-import CohortsOverlapDiagram from '../../../Diagrams/CohortsOverlapDiagram/CohortsOverlapDiagram';
+import PhenotypeHistogram from '../../Diagrams/PhenotypeHistogram/PhenotypeHistogram';
+import CohortsOverlapDiagram from '../../Diagrams/CohortsOverlapDiagram/CohortsOverlapDiagram';
 
 const AttritionTableModal = ({ modalInfo, setModalInfo }) => {
   const modalWidth = 650;
+  const rowIsOutcome = modalInfo.currentCovariateAndCovariatesFromPrecedingRows?.length === 1;
```
Comment (pieterlukasse) [inline src/Analysis/GWASApp/Components/AttritionTableWrapper/AttritionTableModal/AttritionTableModal.jsx:10]: Can we use `rowType` from AttritionTableRow instead? I think that would be a more direct and robust way to determine if row is outcome.

**Review by pieterlukasse — COMMENTED** [review]
thanks. Please see the extra comment above.

**Review by pieterlukasse — APPROVED** [review]
